### DISCUSSION TITLE
RFC 9266: Channel Bindings for TLS 1.3 support

### DISCUSSION THREAD

**Discussion by Neustradamus** [curl/curl#9295] [Ideas]
Can you add the support of RFC 9266: Channel Bindings for TLS 1.3?
- https://datatracker.ietf.org/doc/html/rfc9266 

Little details, to know easily:
- tls-unique for TLS =< 1.2 (RFC5929)
- tls-server-end-point =< 1.2 + 1.3 (RFC5929)
- tls-exporter for TLS = 1.3 (RFC9266)

Thanks in advance.

**Comment by bagder** [comment]
Thanks, but this description sounds as if you're asking for a new feature/change. We use this tracker for bugs and issues only, we put ideas to work on in the future in the TODO document. We basically drown in good ideas so they don't do much use in our tracker.

If you really want to see this happen, start working on an implementation and submit a PR for it or join the mailing list and talk up more interest for it and see what help from others you can get!

**Comment by Neustradamus** [comment]
@bagder: @jas4711 who has added SCRAM support in curl with GNU SASL, has worked on this support in gsasl too.
- https://github.com/curl/curl/pull/6372

**Comment by bagder** [comment]
I'm afraid I don't know what channel bindings are or what the benefits of adding support for them are, so I can't even add a note to the TODO document. Therefore, I cannot offer to do that part.

**Comment by bagder** [comment]
see #7196

**Comment by Neustradamus** [comment]
Attention: RFC 9266 (tls-exporter) is not supported.
It is tls-server-end-point which is used.

Details:
- https://github.com/curl/curl/pull/7196 (@steffen-kiess) + merged commit: https://github.com/curl/curl/commit/0a5ea09a910e7883fd7a1c333e8a36fc782fe537

Linked to:
- https://github.com/curl/curl/issues/9226
- https://github.com/curl/curl/discussions/9295
- https://github.com/curl/curl/pull/13483 (@bk2204) + merged commit: https://github.com/curl/curl/commit/04739054cdac5a0614fb94e3655e313c03399f35
- https://github.com/curl/curl/pull/13098 (@Foorack)
